x/gov/msg.go

```diff
@@ -124,3 +112,47 @@ func (m UpdateElectorateMsg) Validate() error {
 	}
 	return nil
 }
+
+func (CreateElectorateUpdateProposalMsg) Path() string {
+	return pathCreateElectorateUpdateProposalMsg
+}
+
+func (m CreateElectorateUpdateProposalMsg) Validate() error {
+	for i, v := range m.DiffElectors {
+		if v.Weight > maxWeight {
+			return errors.Wrap(errors.ErrInvalidInput, "must not be greater max weight")
+		}
+		if err := v.Address.Validate(); err != nil {
+			return errors.Wrapf(err, "address at position: %d", i)
+		}
+	}
+	return validateCreateProposal(&m)
+}
+
+type commonCreateProposalData interface {
+	GetTitle() string
+	GetDescription() string
+	GetElectorateID() []byte
+	GetStartTime() weave.UnixTime
+	GetAuthor() weave.Address
+}
+
+func validateCreateProposal(m commonCreateProposalData) error {
+	err := m.GetAuthor().Validate()
```

Comment:
This is unusual when reading this code. There is an `err` created here, but it is not checked straight away. Instead, it is (maybe) used later. Weirdly it can be ignored if some earlier validation rule fails.
I think it is much better to keep author validation result close to where it is used. Maybe `switch` is not the right construct to use here?

Reply:
👍 agree